From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Subject: backout git commit c6be45c; breaking sftp with gvfs on maverick

This patch is a direct reversal of the following commit applied on
gvfs git, in GVFS version 1.6.4:

 From c6be45c89346da80d557cdb01756e6e1975ad1ff Mon Sep 17 00:00:00 2001
 From: Tomas Bzatek <tbzatek@redhat.com>
 Date: Mon, 27 Sep 2010 14:10:07 +0000
 Subject: sftp: Use poll() to cope with openssh-5.6 changes

 Patch by Otto Allmendinger
 See bug 629184 for details.

It appears that this patch without openssh-client 5.6 causes sftp calls
to die with connection timeouts.
---
--- a/daemon/gvfsbackendsftp.c	2010-09-27 15:40:14.000000000 -0400
+++ b/daemon/gvfsbackendsftp.c	2010-09-27 15:40:21.368456004 -0400
@@ -24,7 +24,6 @@
 #include <config.h>
 
 #include <stdlib.h>
-#include <sys/poll.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/time.h>
@@ -838,9 +837,10 @@
   GVfsBackendSftp *op_backend = G_VFS_BACKEND_SFTP (backend);
   GInputStream *prompt_stream;
   GOutputStream *reply_stream;
+  fd_set ifds;
+  struct timeval tv;
   int ret;
   int prompt_fd;
-  struct pollfd fds[2];
   char buffer[1024];
   gsize len;
   gboolean aborted = FALSE;
@@ -864,12 +864,14 @@
   ret_val = TRUE;
   while (1)
     {
-      fds[0].fd = stdout_fd;
-      fds[0].events = POLLIN;
-      fds[1].fd = prompt_fd;
-      fds[1].events = POLLIN;
+      FD_ZERO (&ifds);
+      FD_SET (stdout_fd, &ifds);
+      FD_SET (prompt_fd, &ifds);
       
-      ret = poll(fds, 2, SFTP_READ_TIMEOUT);
+      tv.tv_sec = SFTP_READ_TIMEOUT;
+      tv.tv_usec = 0;
+      
+      ret = select (MAX (stdout_fd, prompt_fd)+1, &ifds, NULL, NULL, &tv);
       
       if (ret <= 0)
         {
@@ -880,11 +882,11 @@
           break;
         }
       
-      if (fds[0].revents)
+      if (FD_ISSET (stdout_fd, &ifds))
         break; /* Got reply to initial INIT request */
       
-      if (!(fds[1].revents & POLLIN))
-        continue;
+      g_assert (FD_ISSET (prompt_fd, &ifds));
+      
       
       len = g_input_stream_read (prompt_stream,
                                  buffer, sizeof (buffer) - 1,
